### PR TITLE
Added feature searchbar conjuntion parser

### LINKED ISSUE #4088: Error searching vulnerabilities by the name indexed in the summary
| Wazuh | Elastic | Rev  | Security |
| ----- | ------- | ---- | -------- |
| 4.3   | 7.x     | 4301-1 | Basic, ODFE, Xpack |

| Browser |
| --- |
| Firefox|

### Description

Filter by vulnerability title `CVE-2021-31179 affects Microsoft Office Home and Busines` specified in the VD summary produces a `TypeError`  and `undefined: Error searching items` messages.
This error has been produced using the demo environment.

If we filter manually the name of the vulnerability we can see it in the panel.

VD_OFFICE

### Video

https://user-images.githubusercontent.com/11089305/164710191-5de2e05d-c2ed-44c5-b3ee-b33d281afde9.mp4

### Full traces

- `TypeError: Error fetching items`:

```
_yield$onSearch is undefined
```

- `undefined: Error searching items`

```
3013 - Error in wazuhdb request: Cannot execute SQL query
```

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi Team,
This PR resolves some issues related to `WzSearchBar`, `QInterpreter`, and `filterToObject` method.
All used on Wazuh UI SearchBar.

Closes #4088

### CODE REVIEW

**Review by asteriscos — COMMENTED** [review]
CR: LGTM :heavy_check_mark: 

Test: 
The query is formed correctly, but when typing AND / OR operators, the suggestions don't show properly.
Peek 2022-07-13 17-26

File: public/components/wz-search-bar/components/wz-search-badges/wz-search-badges.tsx [hunk 1]
```diff
@@ -9,31 +9,39 @@
...
   return (
-    <EuiBadge key={key}
-      className='wz-search-badge'
-      color='hollow'
-      iconType='cross'
-      iconSide='right'
-      onFocus={e => e.stopPropagation()}
+    <EuiBadge
+      key={key}
+      data-testid={`${WZ_SEARCH_BADGE_NAME}-${key}`}
```
Comment (Desvelao) [inline public/components/wz-search-bar/components/wz-search-badges/wz-search-badges.tsx:35]: **question**: What this attribute is used for? If it is related to testing, we use `data-test-subj`.

File: public/components/wz-search-bar/lib/q-interpreter.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    it.each([
+      ['and', ';'],
+      [' and', ';'],
+      ['and ', ';'],
+      [' and ', ';'],
+      ['or', ','],
+      [' or', ','],
+      ['or ', ','],
+      [' or ', ','],
+    ])('should translate "%s" conjunto to "%s"', (conjuntion, translated) => {
```
Comment (Desvelao) [inline public/components/wz-search-bar/lib/q-interpreter.test.ts:36]: **nitpick**: did you mean `conjuntion` instead of `conjunto`?

**Review by asteriscos — APPROVED** [review]
Test: :heavy_check_mark: 

Peek 2022-07-14 15-58

Peek 2022-07-14 16-02

File: public/components/wz-search-bar/components/wz-search-badges/wz-search-badges.tsx (hunk 1 above)
Comment (Machi3mfl) [reply public/components/wz-search-bar/components/wz-search-badges/wz-search-badges.tsx:35]: It is used only for testing purposes, using `testing-library/react` and that property I can get the element reference to test it. Only works with data-testid

File: public/components/wz-search-bar/lib/q-interpreter.test.ts (hunk 2 above)
Comment (Machi3mfl) [reply public/components/wz-search-bar/lib/q-interpreter.test.ts:36]: Sorry, i didn't see it and i merged it.

**Pull request merged**